Makefile

```diff
@@ -55,20 +54,24 @@ INC_DIRS += $(UT_CORE_DIR)/src
 
 SRC_DIRS += $(UT_CORE_DIR)/src
 
-XLDFLAGS += -L $(UT_CONTROL)/lib -lut_control -Wl,-rpath-link,$(LIBWEBSOCKET_LIB_DIR)
+XLDFLAGS += -L $(UT_CONTROL)/lib -lut_control
 
 MKDIR_P ?= @mkdir -p
 
-$(info TARGET [$(TARGET)])
-
 # defaults for target arm
 ifeq ($(TARGET),arm)
 CUNIT_VARIANT=arm-rdk-linux-gnueabi
 #CC := arm-rdk-linux-gnueabi-gcc -mthumb -mfpu=vfp -mcpu=cortex-a9 -mfloat-abi=soft -mabi=aapcs-linux -mno-thumb-interwork -ffixed-r8 -fomit-frame-pointer 
 # CFLAGS will be overriden by Caller as required
 INC_DIRS += $(UT_CORE_DIR)/sysroot/usr/include
+TARGET = arm
+XLDFLAGS += -ldl
```

Comment:
Something wrong if you need to use this.. Since ut-core doesn't use dl-load / open functions.

ut-core certainly isn't under arm, something is wrong.


```
#include <dlfcn.h>

int main() {
    void *handle = dlopen("libmyplugin.so", RTLD_LAZY); // Load a shared library
    if (!handle) {
        fprintf(stderr, "Error: %s\n", dlerror()); 
        return 1;
    }

    void (*my_function)() = dlsym(handle, "my_function"); // Get a function from the library
    if (my_function) {
        my_function(); 
    } else {
        fprintf(stderr, "Error: %s\n", dlerror());
    }

    dlclose(handle); // Unload the library
    return 0;
}
```

Reply:
$CC dlopen-test.c 
/home/jpn323/arm-toolchain/sysroots/x86_64-oesdk-linux/usr/libexec/arm-oe-linux-gnueabi/gcc/arm-oe-linux-gnueabi/9.5.0/real-ld: /tmp/ccsrCgdW.o: in function `main':
dlopen-test.c:(.text+0x10): undefined reference to `dlopen'
/home/jpn323/arm-toolchain/sysroots/x86_64-oesdk-linux/usr/libexec/arm-oe-linux-gnueabi/gcc/arm-oe-linux-gnueabi/9.5.0/real-ld: dlopen-test.c:(.text+0x26): undefined reference to `dlerror'
/home/jpn323/arm-toolchain/sysroots/x86_64-oesdk-linux/usr/libexec/arm-oe-linux-gnueabi/gcc/arm-oe-linux-gnueabi/9.5.0/real-ld: dlopen-test.c:(.text+0x4a): undefined reference to `dlsym'
/home/jpn323/arm-toolchain/sysroots/x86_64-oesdk-linux/usr/libexec/arm-oe-linux-gnueabi/gcc/arm-oe-linux-gnueabi/9.5.0/real-ld: dlopen-test.c:(.text+0x68): undefined reference to `dlerror'
/home/jpn323/arm-toolchain/sysroots/x86_64-oesdk-linux/usr/libexec/arm-oe-linux-gnueabi/gcc/arm-oe-linux-gnueabi/9.5.0/real-ld: dlopen-test.c:(.text+0x80): undefined reference to `dlclose'
collect2: error: ld returned 1 exit status

Reply:
links fine with ldl

 $CC dlopen-test.c -ldl
jpn323@janus ~/workspace/gh10-yaml-support-ut-core (feature/gh25-test-group-and-yaml-support)$ 
jpn323@janus ~/workspace/gh10-yaml-support-ut-core (feature/gh25-test-group-and-yaml-support)$ ./a.out 
-bash: ./a.out: cannot execute binary file: Exec format error

Reply:
dlopen-test.c -> Where does this come from?  `ccsrCgdW.o`?

Reply:
ohh..I named the above code you shared [here](https://github.com/rdkcentral/ut-core/pull/86#discussion_r1684379507) to compile and check.